ambry-cloud/src/main/java/com/github/ambry/cloud/RecoveryNetworkClient.java

```diff
@@ -124,134 +149,125 @@ public List<ResponseInfo> sendAndPoll(List<RequestInfo> requestsToSend, Set<Inte
     }
+
+    // return metadata response
     return new ReplicaMetadataResponse(request.getCorrelationId(), request.getClientId(), ServerErrorCode.No_Error,
-        replicaMetadataResponseList, replicaMetadataRequestVersion);
+        responseList, ReplicaMetadataResponse.getCompatibleResponseVersion(request.getVersionId()));
   }
 
   /**
-   * Create {@link MessageInfo} object from {@link CloudBlobMetadata} object.
-   * @param metadata {@link CloudBlobMetadata} object.
+   * Create {@link MessageInfo} object from {@link BlobItem} object.
+   * @param blobItem {@link BlobItem} object.
    * @return {@link MessageInfo} object.
-   * @throws IOException
    */
-  private MessageInfo getMessageInfoFromMetadata(CloudBlobMetadata metadata) throws IOException {
-    BlobId blobId = new BlobId(metadata.getId(), clustermap);
-    long operationTime = (metadata.getDeletionTime() > 0) ? metadata.getDeletionTime()
-        : (metadata.getCreationTime() > 0) ? metadata.getCreationTime() : metadata.getUploadTime();
-    boolean isDeleted = metadata.getDeletionTime() > 0;
-    boolean isTtlUpdated = false;  // No way to know
-    return new MessageInfo(blobId, metadata.getSize(), isDeleted, isTtlUpdated, metadata.getExpirationTime(),
-        (short) metadata.getAccountId(), (short) metadata.getContainerId(), operationTime);
-  }
-
-  protected String getCosmosContinuationToken(String continuationToken) {
-    if (continuationToken == null || continuationToken.isEmpty()) {
-      return null;
-    }
+  private MessageInfo getMessageInfo(BlobItem blobItem) {
+    Map<String, String> metadata = blobItem.getMetadata();
     try {
-      JSONObject continuationTokenJson = new JSONObject(continuationToken);
-      // compositeToken = continuationTokenJson.getString("token");
-      // compositeToken = compositeToken.substring(compositeToken.indexOf('{'), compositeToken.lastIndexOf('}') + 1).replace('\"', '"');
-      return continuationTokenJson.getString("token");
+      /**
+       * Azure blob metadata contains a field expiryTimeMs.
+       * Presence of this field indicates the blob is temporary. ttlUpdated as false in this case.
+       * Absence of this field means the blob was either permanent to start with or,
+       * transitioned from temporary to permanent through a TTL-UPDATE.
+       * There is no way to know which case happened, so just put ttlUpdated as false.
+       * expiryTimeMs is included however and replication-code can decide necessary course of action.
+       * Same goes for undeleted field and lifeVersion is included for replication-code to decide.
+       *
+       * We don't upload CRC during backups. This is probably an issue.
+       */
+      return new MessageInfo(new BlobId(blobItem.getName(), clustermap),
+          Long.parseLong(metadata.get(CloudBlobMetadata.FIELD_SIZE)),
+          metadata.containsKey(CloudBlobMetadata.FIELD_DELETION_TIME),
+          false,
```

Comment:
Do you mind adding the comment for the field we fill in false/null? It'll be easier for reader to know which field it is.

Reply:
yup done